runtime/vm/jvminit.c

```diff
@@ -6519,6 +6526,23 @@ setSignalOptions(J9JavaVM* vm)
 		}
 	}
 
+	argIndex = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOHANDLESIGABRT, NULL);
+	argIndex2 = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXHANDLESIGABRT, NULL);
+
+	if (argIndex2 > argIndex) {
+		/* Enable the JVM abort handler since -XX:+HandleSIGABRT is seen last. */
+		if (OMR_ARE_ALL_BITS_SET(vm->sigFlags, J9_SIG_XRS_SYNC)) {
+			/* Throw error message if both -XX:+HandleSIGABRT and -Xrs/-Xrs:sync are supplied. */
+			j9nls_printf(portLibrary, J9NLS_ERROR, J9NLS_VM_INCOMPATIBLE_CMDLINE_OPTIONS_ERROR, VMOPT_XXHANDLESIGABRT, VMOPT_XRS);
+			return -1;
+		}
+	} else if (argIndex > argIndex2) {
+		/* Disable the JVM abort handler since -XX:-HandleSIGABRT is seen last. */
+		vm->sigFlags |= J9_SIG_NO_SIG_ABRT;
```

Comment:
Does the omr signal code need to be notified about this - ie by `j9sig_set_options`?

Reply:
No, the OMR signal code doesn't need to be notified. The VM handles `SIGABRT` from [rasdump/dmpsup.c](https://github.com/eclipse/openj9/blob/master/runtime/rasdump/dmpsup.c). The option is evaluated on the VM side (`dmpsup.c`) before any of the basic OMR signal functions are invoked. So, OMR doesn't need to be aware of this option.

In OMR, `SIGABRT` is categorized as an `async` signal. A Java user can potentially register a **Java signal handler** for `SIGABRT`: `sun.misc.Signal.handle(...)` -> `JVM_RegisterSignal(...)` -> `omrsig_set_async_signal_handler(...)`.  We don't want to influence this code-path with the new option. This option only enables or disables the VM signal handler for `SIGABRT`, which is maintained in `dmpsup.c`.